scripts/ratchet_coverage/run_coverage.py

```
@@ -0,0 +1,36 @@
+
+def extract_percent(output: str) -> str:
+    match = re.search(r"([0-9]+(?:\.[0-9]+)?)%", output)
+    if not match:
+        typer.echo("Could not parse coverage percent", err=True)
+        raise typer.Exit(code=1)
+    return match[1]
+
+
+def main(
+    args: str = typer.Option("", envvar="INPUT_ARGS"),
+    github_output: Path = typer.Option(..., envvar="GITHUB_OUTPUT"),
```

Ruff (0.11.9)

Do not perform function call `typer.Option` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)